assets/src/data/eventespresso/core/resolvers/relations.js

```diff
@@ -127,9 +127,11 @@ export function* getRelatedEntities( entity, relationModelName ) {
 
 	if ( ! isEmpty( existingEntities ) ) {
 		fullEntities = keepExistingEntitiesInObject(
-			existingEntities.map(
-				( entitiesObject, entityObj ) =>
-					entitiesObject[ entityObj.id ] = entity,
+			existingEntities.reduce(
+				( entitiesObject, entityObj ) => {
+					entitiesObject[ entityObj.id ] = entity;
+					return entitiesObject;
+				},
```

Comment:
This is the actual fix for the issue you discovered @tn3rb.   Crazy stuff and not at all apparent when simply manually reviewing code.  The debugging was showing a new property getting added to TicketEntities of `undefined` and the value was a _DateTimeEntity_ instance!  So I was looking for where that would explicitly be getting set.  Completely missed this in my manual reviews and only surfaced once I sealed the BaseEntity instances.